internal/host/validator.go

```diff
@@ -658,4 +660,180 @@ func (v *validator) printSufficientOrUnknownInstallationDiskSpeed(c *validationC
 	default:
 		return fmt.Sprintf("Unexpected status %s", status.String())
 	}
+
+}
+
+func (v *validator) hasNetworkLatencyRequirementForRole(c *validationContext) ValidationStatus {
+
+	if len(c.cluster.Hosts) == 1 {
+		// Single Node use case
+		return ValidationSuccess
+	}
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending
+	}
+	latency, err := v.validateNetworkLatencyForRole(c.host, c.clusterHostRequirements, c.cluster.Hosts)
+	if err != nil || len(latency) > 0 {
+		return ValidationFailure
+	}
+	return ValidationSuccess
+}
+
+func getHostnameAndRoleByIP(ip string, hosts []*models.Host) (string, models.HostRole, error) {
+
+	for _, h := range hosts {
+		if h.Inventory != "" {
+			inv, err := hostutil.UnmarshalInventory(h.Inventory)
+			if err != nil {
+				return "", "", fmt.Errorf("unable to unmarshall cluster inventory for host %s", h.RequestedHostname)
+			}
+			if strings.Count(ip, ":") < 2 {
+				for _, i := range inv.Interfaces {
+					for _, ipv4 := range i.IPV4Addresses {
+						if ip == ipv4 {
+							return h.RequestedHostname, h.Role, nil
```

Comment:
There is a function to get the hostname here https://github.com/openshift/assisted-service/blob/659b5dab9f676d4bdc70675c20f0b2d1111929ce/internal/host/validator.go#L371 .  Sometimes, the requested hostname is empty

Reply:
That's great! I'll refactor.